spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -189,13 +183,56 @@ def n_atoms(self):
         return self.__n_atoms
 
     @property
-    def _neuron_recorder(self):  # for testing only
+    def incoming_spike_buffer_size(self):
+        return self.__incoming_spike_buffer_size
+
+    @property
+    def parameters(self):
+        return self._parameters
+
+    @property
+    def state_variables(self):
+        return self._state_variables
+
+    @property
+    def neuron_impl(self):
+        return self.__neuron_impl
+
+    @property
+    def n_profile_samples(self):
+        return self.__n_profile_samples
+
+    @property
+    def neuron_recorder(self):  # for testing only
         return self.__neuron_recorder
 
     @property
     def synapse_manager(self):
         return self.__synapse_manager
 
+    def update_state_variables(self):
+        """ processes any changes since init
+
+        :rtype: None
+        """
+
+        # If resetting, reset any state variables that need to be reset
+        if (self.__has_reset_last and
```

Comment:
I would check self.__has_reset_last once and either.
1. return if false
2. Wrap the rest of the black in the if

Reply:
this code is identical to how it was in the app vertex when _write_neuron_parameters was in the app vertex, therefore is out of scope of pr to fix.

do i agree? not sure, as not sure what ya proposing.

Reply:
fixed it